### PR TITLE
API-33204-validate-ssn-from-mpi  

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Adds method to check request ssn against mpi ssn.

## Related issue(s)
- [API-33204](https://jira.devops.va.gov/browse/API-33204)

## Testing done

- Postman

1. Sent POA submit, and put attachment signed in as Tamara, to change Janet Moore's POA
2. repeated above with altered ssn
3. Change the file name in the mockdata repo
Screenshot 2024-01-23 at 10 50 48 AM
4. redis-cli flushall
5. change scope for v1
Screenshot 2024-01-23 at 1 04 02 PM
7. after testing - rename file in mockdata
8. redis-cli flushall again 
9. change scope back, depending

## What areas of the site does it impact?
	modified:   modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb
	modified:   modules/claims_api/spec/requests/v1/power_of_attorney_request_spec.rb

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb [hunk 1]
```diff
@@ -232,8 +233,22 @@ def find_by_ssn(ssn)
           # rubocop:enable Rails/DynamicFindBy
         end
 
+        def check_request_ssn_matches_mpi(req_headers)
+          req_ssn = req_headers['HTTP_X_VA_SSN']
+          ssn = auth_headers['va_eauth_pnid']
```
Comment (acovrig) [inline modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:238]: Is it possible for the pnid to not be the SSN? Should we look at the `pnidtype` field?

Comment (FonzMP) [reply modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:238]: seems like it builds from the EVSS Auth Headers, which takes in the target_veteran - that should come from MPI. 

https://github.com/department-of-veterans-affairs/vets-api/blob/0f08a6f0b24175d7cf37ea0f1f5a63b831783ed0/modules/claims_api/app/controllers/claims_api/v1/forms/base.rb#L38

https://github.com/department-of-veterans-affairs/vets-api/blob/0f08a6f0b24175d7cf37ea0f1f5a63b831783ed0/lib/evss/auth_headers.rb#L29

**Review by acovrig — APPROVED** [review]

**Comment by FonzMP** [comment]
We need this in the submit method as well. We'll be creating an entry in our DB that will never be utilized if it's only added to the PUT for the wet-signed PDF. No reason to create a PowerOfAttorney entry if it will never be possible to use it. We'll just be storing PII indefinitely.

**Comment by FonzMP** [comment]
I can't get a submit to go through locally using mockdata. it reports that it can't find the veteran in MPI if I use an invalid SSN. 🤔

**Comment by stiehlrod** [comment]
> I can't get a submit to go through locally using mockdata. it reports that it can't find the veteran in MPI if I use an invalid SSN. 🤔
> 
> https://github.com/department-of-veterans-affairs/vets-api/blob/4a64e4fd43d2f850c4cecf9c55539b239a1b21cd/modules/claims_api/app/controllers/claims_api/v1/application_controller.rb#L55

Which test user?

**Comment by stiehlrod** [comment]
> I can't get a submit to go through locally using mockdata. it reports that it can't find the veteran in MPI if I use an invalid SSN. 🤔
> 
> https://github.com/department-of-veterans-affairs/vets-api/blob/4a64e4fd43d2f850c4cecf9c55539b239a1b21cd/modules/claims_api/app/controllers/claims_api/v1/application_controller.rb#L55

which test user?

**Comment by FonzMP** [comment]
I'd prefer to have another set of eyes on this since we made the change on a call, but glad to provide my feedback. For any dev testing, you'll have to change the mockdata `mpi/profile` of the Test Veteran you're using to reflect the SSN you're sending in the headers. You can also use a CCG token. glad to walk through it with whoever wants to put eyes on it.

One a sidenote, and the main reason for this comment... [this](https://github.com/department-of-veterans-affairs/vets-api/blob/f8efd81ec3be318aa10ab7b1b5cc83889fc04b36/modules/claims_api/app/controllers/claims_api/v2/application_controller.rb#L37) is a bit reassuring that v2 won't have the same issue.

File: modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb [hunk 2]
```diff
@@ -15,7 +15,10 @@ class PowerOfAttorneyController < ClaimsApi::V1::Forms::Base
           permit_scopes %w[claim.read] if request.get?
         end
         before_action { permit_scopes %w[claim.write] if request.post? || request.put? }
-
+        before_action except: %i[status active] do
+          request&.headers&.to_h.present?
+          @req_headers = request&.headers&.to_h
+        end
```
Comment (FonzMP) [inline modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:21]: could we not use `header_request?`?

Comment (FonzMP) [reply modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb:21]: also, I think the goal is to do something like the `permit_scopes`..... pseudo below

```
before_action { check_request_ssn_matches_mpi except: %w[status active] if header_request? }
```

This currently throws the SSN error on every request without headers.

**Pull request merged**